src/main/java/com/smartbear/readyapi/client/TestRecipeBuilder.java

```diff
@@ -37,27 +54,83 @@ public TestRecipeBuilder withClientCertificatePassword(String password) {
         return this;
     }
 
+    public TestRecipeBuilder withProperties(PropertyBuilder... propertyBuilders) {
+        Arrays.asList(propertyBuilders).forEach(propertyBuilder -> {
+            if (propertyBuilder != null) {
+                this.propertyBuilders.add(propertyBuilder);
+            }
+        });
+        return this;
+    }
+
+    public TestRecipeBuilder withProperty(String key, String value) {
+        this.propertyBuilders.add(property(key, value));
+        return this;
+    }
+
     public TestRecipe buildTestRecipe() {
-        addTestSteps(testCase);
-        return new TestRecipe(testCase);
+        addTestSteps();
+        addProperties();
+        return new TestRecipe(testCase, extractorData);
     }
 
-    private void addTestSteps(TestCase testCase) {
+    private void addProperties() {
+        Map<String, String> propertiesMap = new HashMap<>();
+        propertyBuilders.forEach(propertyBuilder -> {
+            PropertyBuilder.Property property = propertyBuilder.build();
+            propertiesMap.put(property.getKey(), property.getValue());
+        });
+        testCase.setProperties(propertiesMap);
+    }
+
+    private void addTestSteps() {
         List<TestStep> testSteps = new LinkedList<>();
         for (TestStepBuilder testStepBuilder : testStepBuilders) {
             testSteps.add(testStepBuilder.build());
+            if (testStepBuilder instanceof HttpRequestStepBuilder) {
+                List<Extractor> extractorList = ((HttpRequestStepBuilder) testStepBuilder).getExtractors();
+                handleExtractors(extractorList, testSteps);
+            }
         }
         testCase.setTestSteps(testSteps);
     }
 
+    private void handleExtractors(List<Extractor> extractors, List<TestStep> testSteps) {
+        if (!extractors.isEmpty()) {
+            // Add the unique execution key as property, match it in the result report
+            withProperty(ExtractorData.EXTRACTOR_DATA_KEY, extractorData.getExtractorDataId());
+            extractors.forEach(extractor -> {
+                // Base the extractorId on the property if the path is empty, otherwise on the path
+                String extractorId = extractorData
+                        .addExtractorOperator(StringUtils.isEmpty(extractor.getPath()) ?
+                                extractor.getProperty() : extractor.getPath(), extractor.getOperator());
+                withProperty(extractorId, "");
+                testSteps.add(new PropertyTransferTestStepBuilder()
```

Comment:
would it be possible to add just one propertytransfer teststep and then add multiple transfers to it?